cpp/dolfinx/fem/FunctionSpace.h

```diff
@@ -44,7 +44,22 @@ class FunctionSpace
   FunctionSpace(std::shared_ptr<const mesh::Mesh<geometry_type>> mesh,
                 std::shared_ptr<const FiniteElement<geometry_type>> element,
                 std::shared_ptr<const DofMap> dofmap)
-      : _mesh(mesh), _element(element), _dofmap(dofmap),
+      : _mesh(mesh), _dofmaps{dofmap}, _elements{element},
+        _id(boost::uuids::random_generator()()), _root_space_id(_id)
+  {
+    // Do nothing
+  }
+
+  /// @brief Create function space for given mesh, elements and
+  /// degree-of-freedom maps.
+  /// @param[in] mesh Mesh that the space is defined on.
+  /// @param[in] elements Finite element for the space.
```

Comment:
How do we know which element belongs to which cell type?

Reply:
At the moment, they're assumed to be in the same order as `mesh.topology.cell_types`, but this relies on the user passing them in the correct order, so I'll add a check

Reply:
That's fine, but it needs to be documented.